src/engine.rs

```diff
@@ -423,11 +330,20 @@ impl Engine {
         })
     }
 
+    pub fn increment_nonce(address: &Address) {
+        let account_nonce = Self::get_nonce(address);
+        let new_nonce = account_nonce.saturating_add(U256::one());
+        Self::set_nonce(address, &new_nonce);
+    }
+
     #[cfg(feature = "testnet")]
     /// Credits the address with 10 coins from the faucet.
-    pub fn credit(&mut self, address: &Address) -> EngineResult<()> {
-        let new_bal = Self::check_increase_balance(address, &U256::from(10))?;
-        Self::set_balance(address, &new_bal);
+    pub fn credit(&self, address: &Address) -> EngineResult<()> {
+        let balance = Self::get_balance(address);
+        // Saturating adds are intentional
+        let new_balance = balance.saturating_add(U256::from(10));
```

Comment:
This is 10 _Wei_ isn't it? I think 10 _Eth_ would be `10_000_000_000_000_000_000`

As a side note here, we should maybe have some convenience methods and newtypes to prevent these kinds of mistakes going forward (can wait until after launch).

Reply:
I think this is an easy mistake to make, so I've filed #67 for us to remember to make this safer for ourselves later.